components/permissions.py

```diff
@@ -7,8 +7,20 @@
 
 
 class ComponentPermissions(StrictDjangoObjectPermissions):
-    def has_object_permission(self, request: Request, view: APIView, obj: Component) -> bool:
+    def has_object_permission(
+        self, request: Request, view: APIView, obj: Component
+    ) -> bool:
         if obj.status == Component.Status.PUBLIC and request.method in SAFE_METHODS:
             return True
 
         return super().has_object_permission(request, view, obj)
+
+
+class ComponentPrivatePermissions(StrictDjangoObjectPermissions):
```

Comment:
Not sure if this is really needed, but I added it to make sure private components are not filtered out by 
if obj.status == Component.Status.PUBLIC